Source/Fuse.Models/JavaScript.Model.uno

```diff
@@ -0,0 +1,266 @@
+		
+		static void Complete( ModelData md, Visual v )
+		{
+			v.RemoveAllChildren<ModelJavaScript>();
+			
+			//avoid creating without the NameTable as unfortunately UX will set the Model prior to the NameTable
+			if (md.NameTable == null || md.Model == null)
+				return;
+				
+			var parsed = ModelJavaScript.ParseModelExpression(md.Model, md.NameTable);
+			v.Children.Add( new ModelJavaScript(parsed, md.NameTable, null) );
+		}
+
+		static PropertyHandle _modelHandle = Properties.CreateHandle();
+		
+		[UXAttachedPropertySetterAttribute("ModelNameTable")]
+		public static void SetModelNameTable(Visual v, NameTable nt)
+		{
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ NameTable = nt };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.NameTable = nt;
+			}
+			
+			Complete( md, v );
+		}
+
+		//TODO: This should probably be JavaScript.Model, Preview would need to be adjusted as well
+		[UXAttachedPropertySetter("Model"), UXNameScope]
+		public static void SetAppModel(IRootVisualProvider rootVisualProvider, IExpression model)
+		{
+			rootVisualProvider.Root.RemoveAllChildren<ModelJavaScript>();
+
+			var _appModel = ModelJavaScript.CreateFromPreviewState(rootVisualProvider.Root, model);
+			rootVisualProvider.Root.Children.Add(_appModel);
+		}
+
+		internal class ParsedModelExpression
+		{
+			public IExpression Source;
+			public string ModuleName;
+			public string ClassName;
+			public List<string> Args = new List<string>();
+			public List<JavaScript.Dependency> Dependencies = new List<JavaScript.Dependency>();
+			
+			public string ArgString
+			{
+				get
+				{
+					var builder = new StringBuilder();
+					for (int i = 0; i < Args.Count; ++i)
+					{
+						builder.Append(", ");
+						builder.Append(Args[i]);
+					}
+					return builder.ToString();
+				}
+			}
+			
+			public bool CompatibleTo( ParsedModelExpression o )
+			{
+				//there is no way to migrate these now as they might refer to tree objects, thus reject entirely
+				if (Args.Count != 0 || o.Args.Count != 0 ||	
+					Dependencies.Count != 0 || o.Dependencies.Count != 0)
+					return false;
+					
+				return o.ModuleName == ModuleName &&
+					o.ClassName == ClassName;
+			}
+		}
+		
+		internal static ParsedModelExpression ParseModelExpression(IExpression exp, NameTable nt)
+		{
+			var data = exp as Data;
+			if (data != null)
+			{
+				var className = data.Key;
+				return new ParsedModelExpression
+				{
+					ClassName = className,
+					ModuleName = className,
+					Source = exp,
+				};
+			}
+
+			var divide = exp as Divide;
+			if (divide != null)
+			{
+				var left = ParseModelExpression(divide.Left, nt);
+				var right = ParseModelExpression(divide.Right, nt);
+				
+				if (left.Args.Count > 0 || left.Dependencies.Count > 0)
+					throw new Exception( "Invalid Model path expression: " + exp);
+				
+				right.ModuleName = left.ModuleName + "/" + right.ModuleName;
```

Comment:
I've [said this before](https://github.com/fusetools/fuselibs-public/pull/601#discussion_r151416319), but this here looks like a pretty nasty abuse of the expression-system to me... I'm rather worried that needs of the expression system and model-paths might diverge in the future, rendering a solution like this unfeasible. Don't we have some other options here, like `[UXVerbatim]`?